File: backend/src/main/java/com/tutict/finalassignmentbackend/entity/AppealRecord.java
package com.tutict.finalassignmentbackend.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 申诉记录表实体类，对应数据库表 "appeal_record"
 * 记录对违法记录的申诉信息
 */
@Data
@TableName("appeal_record")
public class AppealRecord implements Serializable {
    /**
     * 序列化版本 UID
     */
    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 申诉记录 ID，主键，自动增长
     */
    @TableId(value = "appeal_id", type = IdType.AUTO)
    private Long appealId;

    /**
     * 违法记录 ID
     */
    @TableField("offense_id")
    private Long offenseId;

    /**
     * 申诉编号
     */
    @TableField("appeal_number")
    private String appealNumber;

    /**
     * 申诉人姓名
     */
    @TableField("appellant_name")
    private String appellantName;

    /**
     * 申诉人身份证号
     */
    @TableField("appellant_id_card")
    private String appellantIdCard;

    /**
     * 申诉人联系电话
     */
    @TableField("appellant_contact")
    private String appellantContact;

    /**
     * 申诉人电子邮箱
     */
    @TableField("appellant_email")
    private String appellantEmail;

    /**
     * 申诉人联系地址
     */
    @TableField("appellant_address")
    private String appellantAddress;

    /**
     * 申诉类型 (Information_Error, Equipment_Error, Judgment_Error, Force_Majeure, Other)
     */
    @TableField("appeal_type")
    private String appealType;

    /**
     * 申诉理由
     */
    @TableField("appeal_reason")
    private String appealReason;

    /**
     * 申诉时间
     */
    @TableField("appeal_time")
    private LocalDateTime appealTime;

    /**
     * 证据说明
     */
    @TableField("evidence_description")
    private String evidenceDescription;

    /**
     * 证据文件 URL 列表 (JSON数组)
     */
    @TableField("evidence_urls")
    private String evidenceUrls;

    /**
     * 受理状态 (Pending, Accepted, Rejected, Need_Supplement)
     */
    @TableField("acceptance_status")
    private String acceptanceStatus;

    /**
     * 受理时间
     */
    @TableField("acceptance_time")
    private LocalDateTime acceptanceTime;

    /**
     * 受理人
     */
    @TableField("acceptance_handler")
    private String acceptanceHandler;

    /**
     * 不予受理原因
     */
    @TableField("rejection_reason")
    private String rejectionReason;

    /**
     * 处理状态 (Unprocessed, Under_Review, Approved, Rejected, Withdrawn)
     */
    @TableField("process_status")
    private String processStatus;

    /**
     * 处理时间
     */
    @TableField("process_time")
    private LocalDateTime processTime;

    /**
     * 处理结果
     */
    @TableField("process_result")
    private String processResult;

    /**
     * 处理人
     */
    @TableField("process_handler")
    private String processHandler;

    /**
     * 创建时间
     */
    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /**
     * 创建人
     */
    @TableField("created_by")
    private String createdBy;

    /**
     * 更新人
     */
    @TableField("updated_by")
    private String updatedBy;

    /**
     * 软删除时间戳
     */
    @TableField("deleted_at")
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;

    /**
     * 备注
     */
    @TableField("remarks")
    private String remarks;
}

File: flutter_app/lib/features/model/appeal_record.dart
class AppealRecordModel {
  final int? appealId;
  final int? offenseId;
  final String? appealNumber;
  final String? appellantName;
  final String? appellantIdCard;
  final String? appellantContact;
  final String? appellantEmail;
  final String? appellantAddress;
  final String? appealType;
  final String? appealReason;
  final DateTime? appealTime;
  final String? evidenceDescription;
  final String? evidenceUrls;
  final String? acceptanceStatus;
  final DateTime? acceptanceTime;
  final String? acceptanceHandler;
  final String? rejectionReason;
  final String? processStatus;
  final DateTime? processTime;
  final String? processResult;
  final String? processHandler;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const AppealRecordModel({
    this.appealId,
    this.offenseId,
    this.appealNumber,
    this.appellantName,
    this.appellantIdCard,
    this.appellantContact,
    this.appellantEmail,
    this.appellantAddress,
    this.appealType,
    this.appealReason,
    this.appealTime,
    this.evidenceDescription,
    this.evidenceUrls,
    this.acceptanceStatus,
    this.acceptanceTime,
    this.acceptanceHandler,
    this.rejectionReason,
    this.processStatus,
    this.processTime,
    this.processResult,
    this.processHandler,
    this.createdAt,
    this.updatedAt,
  });

  factory AppealRecordModel.fromJson(Map<String, dynamic> json) {
    return AppealRecordModel(
      appealId: json['appealId'],
      offenseId: json['offenseId'],
      appealNumber: json['appealNumber'],
      appellantName: json['appellantName'],
      appellantIdCard: json['appellantIdCard'],
      appellantContact: json['appellantContact'],
      appellantEmail: json['appellantEmail'],
      appellantAddress: json['appellantAddress'],
      appealType: json['appealType'],
      appealReason: json['appealReason'],
      appealTime:
          json['appealTime'] != null ? DateTime.tryParse(json['appealTime']) : null,
      evidenceDescription: json['evidenceDescription'],
      evidenceUrls: json['evidenceUrls'],
      acceptanceStatus: json['acceptanceStatus'],
      acceptanceTime: json['acceptanceTime'] != null
          ? DateTime.tryParse(json['acceptanceTime'])
          : null,
      acceptanceHandler: json['acceptanceHandler'],
      rejectionReason: json['rejectionReason'],
      processStatus: json['processStatus'],
      processTime:
          json['processTime'] != null ? DateTime.tryParse(json['processTime']) : null,
      processResult: json['processResult'],
      processHandler: json['processHandler'],
      createdAt:
          json['createdAt'] != null ? DateTime.tryParse(json['createdAt']) : null,
      updatedAt:
          json['updatedAt'] != null ? DateTime.tryParse(json['updatedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'appealId': appealId,
        'offenseId': offenseId,
        'appealNumber': appealNumber,
        'appellantName': appellantName,
        'appellantIdCard': appellantIdCard,
        'appellantContact': appellantContact,
        'appellantEmail': appellantEmail,
        'appellantAddress': appellantAddress,
        'appealType': appealType,
        'appealReason': appealReason,
        'appealTime': appealTime?.toIso8601String(),
        'evidenceDescription': evidenceDescription,
        'evidenceUrls': evidenceUrls,
        'acceptanceStatus': acceptanceStatus,
        'acceptanceTime': acceptanceTime?.toIso8601String(),
        'acceptanceHandler': acceptanceHandler,
        'rejectionReason': rejectionReason,
        'processStatus': processStatus,
        'processTime': processTime?.toIso8601String(),
        'processResult': processResult,
        'processHandler': processHandler,
        'createdAt': createdAt?.toIso8601String(),
        'updatedAt': updatedAt?.toIso8601String(),
      };

  AppealRecordModel copyWith({
    int? appealId,
    int? offenseId,
    String? appealNumber,
    String? appellantName,
    String? appellantIdCard,
    String? appellantContact,
    String? appellantEmail,
    String? appellantAddress,
    String? appealType,
    String? appealReason,
    DateTime? appealTime,
    String? evidenceDescription,
    String? evidenceUrls,
    String? acceptanceStatus,
    DateTime? acceptanceTime,
    String? acceptanceHandler,
    String? rejectionReason,
    String? processStatus,
    DateTime? processTime,
    String? processResult,
    String? processHandler,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return AppealRecordModel(
      appealId: appealId ?? this.appealId,
      offenseId: offenseId ?? this.offenseId,
      appealNumber: appealNumber ?? this.appealNumber,
      appellantName: appellantName ?? this.appellantName,
      appellantIdCard: appellantIdCard ?? this.appellantIdCard,
      appellantContact: appellantContact ?? this.appellantContact,
      appellantEmail: appellantEmail ?? this.appellantEmail,
      appellantAddress: appellantAddress ?? this.appellantAddress,
      appealType: appealType ?? this.appealType,
      appealReason: appealReason ?? this.appealReason,
      appealTime: appealTime ?? this.appealTime,
      evidenceDescription: evidenceDescription ?? this.evidenceDescription,
      evidenceUrls: evidenceUrls ?? this.evidenceUrls,
      acceptanceStatus: acceptanceStatus ?? this.acceptanceStatus,
      acceptanceTime: acceptanceTime ?? this.acceptanceTime,
      acceptanceHandler: acceptanceHandler ?? this.acceptanceHandler,
      rejectionReason: rejectionReason ?? this.rejectionReason,
      processStatus: processStatus ?? this.processStatus,
      processTime: processTime ?? this.processTime,
      processResult: processResult ?? this.processResult,
      processHandler: processHandler ?? this.processHandler,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
